### ISSUE TITLE
AtomicRegisterCoordinatorTests.testNodesJoinAfterStableCluster failure

### ISSUE THREAD

**Issue by astefan** [elastic/elasticsearch#95888]
### CI Link

https://gradle-enterprise.elastic.co/s/44beutx3g3bh4

### Repro line

`gradlew ':server:test' --tests "org.elasticsearch.cluster.coordination.AtomicRegisterCoordinatorTests.testNodesJoinAfterStableCluster" -Dtests.seed=A15E00EB07D6B140 -Dtests.locale=nl -Dtests.timezone=Africa/Dakar -Druntime.java=20`

### Does it reproduce?

Yes

### Applicable branches

8.8

### Failure excerpt

Similar issue in https://github.com/elastic/elasticsearch/issues/95887

```
Suite: Test class org.elasticsearch.cluster.coordination.AtomicRegisterCoordinatorTests
  2> REPRODUCE WITH: gradlew ':server:test' --tests "org.elasticsearch.cluster.coordination.AtomicRegisterCoordinatorTests.testNodesJoinAfterStableCluster" -Dtests.seed=A15E00EB07D6B140 -Dtests.locale=nl -Dtests.timezone=Africa/Dakar -Druntime.java=20
  2> java.lang.AssertionError: node1 has no NO_MASTER_BLOCK
    Expected: <false>
         but: was <true>
        at __randomizedtesting.SeedInfo.seed([A15E00EB07D6B140:9F59B6583AADE485]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.elasticsearch.cluster.coordination.AbstractCoordinatorTestCase$Cluster.stabilise(AbstractCoordinatorTestCase.java:583)
        at org.elasticsearch.cluster.coordination.AbstractCoordinatorTestCase$Cluster.stabilise(AbstractCoordinatorTestCase.java:553)
        at org.elasticsearch.cluster.coordination.CoordinatorTests.testNodesJoinAfterStableCluster(CoordinatorTests.java:212)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
        at java.base/java.lang.reflect.Method.invoke(Method.java:578)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
        at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
        at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
        at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
        at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
        at java.base/java.lang.Thread.run(Thread.java:1623)
  1> [2023-05-05T11:13:32,808][WARN ][o.e.c.c.ClusterFormationFailureHelper] [testWarnLoggingOnRegisterFailures{nodeId={node0}{9B4ZXgAAQAC-t2F-AAAAAA}}] master not discovered or elected yet, an election requires a node with id [node0], have discovered possible quorum [{node0}{9B4ZXgAAQAC-t2F-AAAAAA}{0.0.0.0}{0.0.0.0:2}{IScdfhilmrstw}{8.8.0}]; discovery will continue using [0.0.0.0:2] from hosts providers and [{node0}{9B4ZXgAAQAC-t2F-AAAAAA}{0.0.0.0}{0.0.0.0:2}{IScdfhilmrstw}{8.8.0}] from last-known cluster state; node term 0, last-accepted version 0 in term 0; for troubleshooting guidance, see https://www.elastic.co/guide/en/elasticsearch/reference/master/discovery-troubleshooting.html
  1> [2023-05-05T11:13:32,811][WARN ][o.e.c.c.ClusterFormationFailureHelper] [testWarnLoggingOnRegisterFailures{nodeId={node0}{9B4ZXgAAQAC-t2F-AAAAAA}}] master not discovered or elected yet, an election requires a node with id [node0], have discovered possible quorum [{node0}{9B4ZXgAAQAC-t2F-AAAAAA}{0.0.0.0}{0.0.0.0:2}{IScdfhilmrstw}{8.8.0}]; discovery will continue using [0.0.0.0:2] from hosts providers and [{node0}{9B4ZXgAAQAC-t2F-AAAAAA}{0.0.0.0}{0.0.0.0:2}{IScdfhilmrstw}{8.8.0}] from last-known cluster state; node term 0, last-accepted version 0 in term 0; for troubleshooting guidance, see https://www.elastic.co/guide/en/elasticsearch/reference/master/discovery-troubleshooting.html
  1> [2023-05-05T11:13:32,812][WARN ][o.e.c.c.ClusterFormationFailureHelper] [testWarnLoggingOnRegisterFailures{nodeId={node0}{9B4ZXgAAQAC-t2F-AAAAAA}}] master not discovered or elected yet, an election requires a node with id [node0], have discovered possible quorum [{node0}{9B4ZXgAAQAC-t2F-AAAAAA}{0.0.0.0}{0.0.0.0:2}{IScdfhilmrstw}{8.8.0}]; discovery will continue using [0.0.0.0:2] from hosts providers and [{node0}{9B4ZXgAAQAC-t2F-AAAAAA}{0.0.0.0}{0.0.0.0:2}{IScdfhilmrstw}{8.8.0}] from last-known cluster state; node term 3, last-accepted version 0 in term 0; joining [{node0}{9B4ZXgAAQAC-t2F-AAAAAA}{0.0.0.0}{0.0.0.0:2}{IScdfhilmrstw}{8.8.0}] in term [3] has status [waiting for response] after [15.2s/15276ms]; for troubleshooting guidance, see https://www.elastic.co/guide/en/elasticsearch/reference/master/discovery-troubleshooting.html
  1> [2023-05-05T11:13:32,812][WARN ][o.e.c.c.ClusterFormationFailureHelper] [testWarnLoggingOnRegisterFailures{nodeId={node0}{9B4ZXgAAQAC-t2F-AAAAAA}}] master not discovered or elected yet, an election requires a node with id [node0], have discovered possible quorum [{node0}{9B4ZXgAAQAC-t2F-AAAAAA}{0.0.0.0}{0.0.0.0:2}{IScdfhilmrstw}{8.8.0}]; discovery will continue using [0.0.0.0:2] from hosts providers and [{node0}{9B4ZXgAAQAC-t2F-AAAAAA}{0.0.0.0}{0.0.0.0:2}{IScdfhilmrstw}{8.8.0}] from last-known cluster state; node term 3, last-accepted version 0 in term 0; joining [{node0}{9B4ZXgAAQAC-t2F-AAAAAA}{0.0.0.0}{0.0.0.0:2}{IScdfhilmrstw}{8.8.0}] in term [3] has status [waiting for response] after [30.7s/30748ms]; for troubleshooting guidance, see https://www.elastic.co/guide/en/elasticsearch/reference/master/discovery-troubleshooting.html
  1> [2023-05-05T11:13:32,812][WARN ][o.e.c.c.ClusterFormationFailureHelper] [testWarnLoggingOnRegisterFailures{nodeId={node0}{9B4ZXgAAQAC-t2F-AAAAAA}}] master not discovered or elected yet, an election requires a node with id [node0], have discovered possible quorum [{node0}{9B4ZXgAAQAC-t2F-AAAAAA}{0.0.0.0}{0.0.0.0:2}{IScdfhilmrstw}{8.8.0}]; discovery will continue using [0.0.0.0:2] from hosts providers and [{node0}{9B4ZXgAAQAC-t2F-AAAAAA}{0.0.0.0}{0.0.0.0:2}{IScdfhilmrstw}{8.8.0}] from last-known cluster state; node term 5, last-accepted version 0 in term 0; joining [{node0}{9B4ZXgAAQAC-t2F-AAAAAA}{0.0.0.0}{0.0.0.0:2}{IScdfhilmrstw}{8.8.0}] in term [5] has status [waiting for local cluster applier] after [0ms]; for troubleshooting guidance, see https://www.elastic.co/guide/en/elasticsearch/reference/master/discovery-troubleshooting.html
  1> [2023-05-05T11:13:32,813][WARN ][o.e.c.c.ClusterFormationFailureHelper] [testWarnLoggingOnRegisterFailures{nodeId={node0}{9B4ZXgAAQAC-t2F-AAAAAA}}] master not discovered or elected yet, an election requires a node with id [node0], have discovered possible quorum [{node0}{9B4ZXgAAQAC-t2F-AAAAAA}{0.0.0.0}{0.0.0.0:2}{IScdfhilmrstw}{8.8.0}]; discovery will continue using [0.0.0.0:2] from hosts providers and [{node0}{9B4ZXgAAQAC-t2F-AAAAAA}{0.0.0.0}{0.0.0.0:2}{IScdfhilmrstw}{8.8.0}] from last-known cluster state; node term 5, last-accepted version 0 in term 0; joining [{node0}{9B4ZXgAAQAC-t2F-AAAAAA}{0.0.0.0}{0.0.0.0:2}{IScdfhilmrstw}{8.8.0}] in term [5] has status [waiting for local cluster applier] after [13.4s/13462ms]; for troubleshooting guidance, see https://www.elastic.co/guide/en/elasticsearch/reference/master/discovery-troubleshooting.html
  1> [2023-05-05T11:13:32,822][WARN ][o.e.c.c.ClusterFormationFailureHelper] [testWarnLoggingOnRegisterFailures{nodeId={node0}{lhPIuv__T_-wZrdQAAAAAA}}] master not discovered or elected yet, an election requires a node with id [node0], have discovered possible quorum [{node0}{lhPIuv__T_-wZrdQAAAAAA}{0.0.0.0}{0.0.0.0:3}{IScdfhilmrstw}{8.8.0}]; discovery will continue using [0.0.0.0:3] from hosts providers and [{node0}{lhPIuv__T_-wZrdQAAAAAA}{0.0.0.0}{0.0.0.0:3}{IScdfhilmrstw}{8.8.0}] from last-known cluster state; node term 8, last-accepted version 0 in term 0; joining [{node0}{lhPIuv__T_-wZrdQAAAAAA}{0.0.0.0}{0.0.0.0:3}{IScdfhilmrstw}{8.8.0}] in term [8] has status [waiting for local cluster applier] after [26.8s/26862ms]; for troubleshooting guidance, see https://www.elastic.co/guide/en/elasticsearch/reference/master/discovery-troubleshooting.html
  1> [2023-05-05T11:13:32,830][WARN ][o.e.c.c.ClusterFormationFailureHelper] [testWarnLoggingOnRegisterFailures{nodeId={node0}{3EZ9awAAQAC-bCEgAAAAAA}}] master not discovered or elected yet, an election requires a node with id [node0], have discovered possible quorum [{node0}{3EZ9awAAQAC-bCEgAAAAAA}{0.0.0.0}{0.0.0.0:4}{IScdfhilmrstw}{8.8.0}]; discovery will continue using [] from hosts providers and [{node0}{3EZ9awAAQAC-bCEgAAAAAA}{0.0.0.0}{0.0.0.0:4}{IScdfhilmrstw}{8.8.0}] from last-known cluster state; node term 0, last-accepted version 0 in term 0; for troubleshooting guidance, see https://www.elastic.co/guide/en/elasticsearch/reference/master/discovery-troubleshooting.html
  1> [2023-05-05T11:13:32,830][WARN ][o.e.c.c.s.StoreHeartbeatService] [testWarnLoggingOnRegisterFailures{nodeId={node0}{3EZ9awAAQAC-bCEgAAAAAA}}] failed to read heartbeat from store
  1> java.io.IOException: simulating disrupted access to shared store
  1> 	at org.elasticsearch.cluster.coordination.stateless.DisruptibleHeartbeatStore.readLatestHeartbeat(DisruptibleHeartbeatStore.java:36) ~[framework-8.8.0-SNAPSHOT.jar:?]
  1> 	at org.elasticsearch.cluster.coordination.stateless.StoreHeartbeatService.runIfNoRecentLeader(StoreHeartbeatService.java:99) ~[main/:?]
  1> 	at org.elasticsearch.cluster.coordination.stateless.AtomicRegisterPreVoteCollector.start(AtomicRegisterPreVoteCollector.java:30) ~[main/:?]
  1> 	at org.elasticsearch.cluster.coordination.Coordinator$9.run(Coordinator.java:1705) ~[main/:?]
  1> 	at org.elasticsearch.cluster.coordination.ElectionSchedulerFactory$ElectionScheduler$1.doRun(ElectionSchedulerFactory.java:195) ~[main/:?]
  1> 	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:983) ~[main/:?]
  1> 	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26) ~[main/:?]
  1> 	at org.elasticsearch.common.util.concurrent.DeterministicTaskQueue$2$2.run(DeterministicTaskQueue.java:368) ~[framework-8.8.0-SNAPSHOT.jar:?]
  1> 	at org.elasticsearch.common.util.concurrent.DeterministicTaskQueue$3.run(DeterministicTaskQueue.java:534) ~[framework-8.8.0-SNAPSHOT.jar:?]
  1> 	at org.elasticsearch.cluster.coordination.AbstractCoordinatorTestCase$Cluster$ClusterNode$2.run(AbstractCoordinatorTestCase.java:1261) ~[framework-8.8.0-SNAPSHOT.jar:?]
  1> 	at org.elasticsearch.common.util.concurrent.DeterministicTaskQueue.runTask(DeterministicTaskQueue.java:135) ~[framework-8.8.0-SNAPSHOT.jar:?]
  1> 	at org.elasticsearch.common.util.concurrent.DeterministicTaskQueue.runRandomTask(DeterministicTaskQueue.java:129) ~[framework-8.8.0-SNAPSHOT.jar:?]
  1> 	at org.elasticsearch.cluster.coordination.AbstractCoordinatorTestCase$Cluster.runRandomly(AbstractCoordinatorTestCase.java:514) ~[framework-8.8.0-SNAPSHOT.jar:?]
  1> 	at org.elasticsearch.cluster.coordination.AbstractCoordinatorTestCase$Cluster.runRandomly(AbstractCoordinatorTestCase.java:368) ~[framework-8.8.0-SNAPSHOT.jar:?]
  1> 	at org.elasticsearch.cluster.coordination.AtomicRegisterCoordinatorTests.testWarnLoggingOnRegisterFailures(AtomicRegisterCoordinatorTests.java:129) ~[test/:?]
  1> 	at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104) ~[?:?]
  1> 	at java.lang.reflect.Method.invoke(Method.java:578) ~[?:?]
  1> 	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758) ~[randomizedtesting-runner-2.8.0.jar:?]
  1> 	at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946) ~[randomizedtesting-runner-2.8.0.jar:?]
  1> 	at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982) ~[randomizedtesting-runner-2.8.0.jar:?]
  1> 	at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996) ~[randomizedtesting-runner-2.8.0.jar:?]
  1> 	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) ~[randomizedtesting-runner-2.8.0.jar:?]
  1> 	at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48) ~[lucene-test-framework-9.6.0-snapshot-246ac4bcbe6.jar:9.6.0-snapshot-246ac4bcbe6 246ac4bcbe6c0ea7bebcc4f90b2be0536f00d842 - 2023-04-27 08:02:31]
  1> 	at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43) ~[lucene-test-framework-9.6.0-snapshot-246ac4bcbe6.jar:9.6.0-snapshot-246ac4bcbe6 246ac4bcbe6c0ea7bebcc4f90b2be0536f00d842 - 2023-04-27 08:02:31]
  1> 	at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45) ~[lucene-test-framework-9.6.0-snapshot-246ac4bcbe6.jar:9.6.0-snapshot-246ac4bcbe6 246ac4bcbe6c0ea7bebcc4f90b2be0536f00d842 - 2023-04-27 08:02:31]
  1> 	at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60) ~[lucene-test-framework-9.6.0-snapshot-246ac4bcbe6.jar:9.6.0-snapshot-246ac4bcbe6 246ac4bcbe6c0ea7bebcc4f90b2be0536f00d842 - 2023-04-27 08:02:31]
  1> 	at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44) ~[lucene-test-framework-9.6.0-snapshot-246ac4bcbe6.jar:9.6.0-snapshot-246ac4bcbe6 246ac4bcbe6c0ea7bebcc4f90b2be0536f00d842 - 2023-04-27 08:02:31]
  1> 	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) ~[randomizedtesting-runner-2.8.0.jar:?]
  1> 	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390) ~[randomizedtesting-runner-2.8.0.jar:?]
  1> 	at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843) ~[randomizedtesting-runner-2.8.0.jar:?]
  1> 	at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490) ~[randomizedtesting-runner-2.8.0.jar:?]
  1> 	at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955) ~[randomizedtesting-runner-2.8.0.jar:?]
  1> 	at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840) ~[randomizedtesting-runner-2.8.0.jar:?]
  1> 	at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891) ~[randomizedtesting-runner-2.8.0.jar:?]
  1> 	at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902) ~[randomizedtesting-runner-2.8.0.jar:?]
  1> 	at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43) ~[lucene-test-framework-9.6.0-snapshot-246ac4bcbe6.jar:9.6.0-snapshot-246ac4bcbe6 246ac4bcbe6c0ea7bebcc4f90b2be0536f00d842 - 2023-04-27 08:02:31]
  1> 	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) ~[randomizedtesting-runner-2.8.0.jar:?]
  1> 	at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38) ~[lucene-test-framework-9.6.0-snapshot-246ac4bcbe6.jar:9.6.0-snapshot-246ac4bcbe6 246ac4bcbe6c0ea7bebcc4f90b2be0536f00d842 - 2023-04-27 08:02:31]
  1> 	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40) ~[randomizedtesting-runner-2.8.0.jar:?]
  1> 	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40) ~[randomizedtesting-runner-2.8.0.jar:?]
  1> 	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) ~[randomizedtesting-runner-2.8.0.jar:?]
  1> 	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) ~[randomizedtesting-runner-2.8.0.jar:?]
  1> 	at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53) ~[lucene-test-framework-9.6.0-snapshot-246ac4bcbe6.jar:9.6.0-snapshot-246ac4bcbe6 246ac4bcbe6c0ea7bebcc4f90b2be0536f00d842 - 2023-04-27 08:02:31]
  1> 	at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43) ~[lucene-test-framework-9.6.0-snapshot-246ac4bcbe6.jar:9.6.0-snapshot-246ac4bcbe6 246ac4bcbe6c0ea7bebcc4f90b2be0536f00d842 - 2023-04-27 08:02:31]
  1> 	at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44) ~[lucene-test-framework-9.6.0-snapshot-246ac4bcbe6.jar:9.6.0-snapshot-246ac4bcbe6 246ac4bcbe6c0ea7bebcc4f90b2be0536f00d842 - 2023-04-27 08:02:31]
  1> 	at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60) ~[lucene-test-framework-9.6.0-snapshot-246ac4bcbe6.jar:9.6.0-snapshot-246ac4bcbe6 246ac4bcbe6c0ea7bebcc4f90b2be0536f00d842 - 2023-04-27 08:02:31]
  1> 	at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47) ~[lucene-test-framework-9.6.0-snapshot-246ac4bcbe6.jar:9.6.0-snapshot-246ac4bcbe6 246ac4bcbe6c0ea7bebcc4f90b2be0536f00d842 - 2023-04-27 08:02:31]
  1> 	at com.carrotsearch.randomizedtesting.rules.Statement  1> Adapter.evaluate(StatementAdapter.java:36) ~[randomizedtesting-runner-2.8.0.jar:?]
  1> 	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390) ~[randomizedtesting-runner-2.8.0.jar:?]
  1> 	at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850) ~[randomizedtesting-runner-2.8.0.jar:?]
  1> 	at java.lang.Thread.run(Thread.java:1623) ~[?:?]
```

**Comment by idegtiarenko** [comment]
The failure seems similar to https://github.com/elastic/elasticsearch/issues/96273 and is not reproduces on 8.9 branch.
I wonder if this was fixed by https://github.com/elastic/elasticsearch/pull/96490?

**Comment by idegtiarenko** [comment]
I was able to confirm that applying https://github.com/elastic/elasticsearch/pull/96490 fixes this issue on 8.8

**Comment by idegtiarenko** [comment]
I am closing it as a root issue is fixed in 8.9+ versions and we are not planning any new 8.8.x
